pkg/service/manifest/manifest.go

```diff
@@ -186,59 +179,57 @@ func (s Service) SubmitApplication(request SubmitApplicationRequest) (*SubmitApp
 	}
 
 	// build the credential response
+	// TODO(gabe) need to check if this can be fulfilled and conditionally return success/denial
 	responseBuilder := manifest.NewCredentialResponseBuilder(request.Application.ManifestID)
-	responseBuilder.SetApplicationID(credApp.ID)
-	responseBuilder.SetFulfillment(credApp.PresentationSubmission.DescriptorMap)
-
-	credRes, err := responseBuilder.Build()
-	if err != nil {
-		errMsg := "could not build response"
-		return nil, util.LoggingErrorMsg(err, errMsg)
+	if err := responseBuilder.SetApplicationID(credApp.ID); err != nil {
+		return nil, util.LoggingErrorMsg(err, "could not fulfill credential application: could not set application id")
 	}
 
-	// store the response
-	responseStorageRequest := manifeststorage.StoredResponse{
-		ID:         credRes.ID,
-		Response:   *credRes,
-		ManifestID: request.Application.ManifestID,
-	}
-
-	var creds []credential.VerifiableCredential
+	var creds []credsdk.VerifiableCredential
 	for _, od := range gotManifest.Manifest.OutputDescriptors {
-
-		if err := s.manifestStorage.StoreResponse(responseStorageRequest); err != nil {
-			errMsg := "could not store response"
-			return nil, util.LoggingErrorMsg(err, errMsg)
+		credentialRequest := credential.CreateCredentialRequest{
+			Issuer:     gotManifest.Manifest.Issuer.ID,
+			Subject:    request.RequesterDID,
+			JSONSchema: od.Schema,
+			// TODO(gabe) need to add in data here to match the request + schema
+			Data: map[string]interface{}{},
 		}
 
-		credentialBuilder := credential.NewVerifiableCredentialBuilder()
-		credentialBuilder.SetIssuer(gotManifest.Manifest.Issuer.ID)
-		credentialBuilder.SetCredentialSubject(map[string]interface{}{
-			"id": request.RequesterDID,
-		})
-		credentialBuilder.SetIssuanceDate(time.Now().Format(time.RFC3339))
-
-		cred, err := credentialBuilder.Build()
+		createdResponse, err := s.credential.CreateCredential(credentialRequest)
```

Comment:
This probably needs to be credentialResponse

Reply:
done